Remotion/ObjectBinding/Web/UI/Controls/BocListImplementation/Rendering/BocColumnRendererBase.cs

```diff
@@ -355,8 +363,80 @@ private void RenderDataCell (BocColumnRenderingContext<TBocColumnDefinition> ren
+        iconInfo.Render(renderingContext.Writer, renderingContext.Control);
+
+        renderingContext.Writer.RenderEndTag();
+      }
+      else if (arguments.ColumnsWithValidationFailures[renderingContext.ColumnIndex])
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, CssClasses.ValidationErrorMarker);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        var iconInfo = new IconInfo(_resourceUrlFactory.CreateThemedResourceUrl(typeof(InfrastructureResourceUrlFactory), ResourceType.Image, c_validationIndicatorPlaceholderIcon).GetUrl());
+        iconInfo.Render(renderingContext.Writer, renderingContext.Control);
+
+        renderingContext.Writer.RenderEndTag();
+      }
+
       RenderCellContents(renderingContext, arguments);
 
+      if (validationFailures.Count > 0)
+      {
+        renderingContext.Writer.AddAttribute(HtmlTextWriterAttribute.Class, _cssClasses.CssClassScreenReaderText);
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+        renderingContext.Writer.WriteEncodedText(validationInfoText);
+        renderingContext.Writer.RenderEndTag();
+
+        renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Ul);
+        foreach (var validationFailure in validationFailures)
+        {
+          if (_renderingFeatures.EnableDiagnosticMetadata)
+          {
+            var validatedObjectWithIdentity = validationFailure.Failure.ValidatedObject as IBusinessObjectWithIdentity;
+            var rowObjectWithIdentity = validationFailure.RowObject as IBusinessObjectWithIdentity;
+
+            renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceRow, rowObjectWithIdentity?.UniqueIdentifier ?? string.Empty);
+            renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.BocListValidationFailureSourceColumn, validationFailure.ColumnDefinition?.ItemID ?? string.Empty);
+            renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.ValidationFailureSourceBusinessObject, validatedObjectWithIdentity?.UniqueIdentifier ?? string.Empty);
+            renderingContext.Writer.AddAttribute(DiagnosticMetadataAttributesForObjectBinding.ValidationFailureSourceProperty, validationFailure.Failure.ValidatedProperty?.Identifier ?? string.Empty);
+          }
+
+          renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Li);
+          renderingContext.Writer.RenderBeginTag(HtmlTextWriterTag.Span);
+          renderingContext.Writer.WriteEncodedText(validationFailure.Failure.ErrorMessage);
+          renderingContext.Writer.RenderEndTag();
+          renderingContext.Writer.RenderEndTag();
+        }
+
+        renderingContext.Writer.RenderEndTag();
+
+        renderingContext.Writer.RenderEndTag();
```

Comment:
I usually add comments on which element gets closed when there's more than one item in the method. makes for easier troubleshooting / review / maintenance.